answer_rocket/skill.py

```diff
@@ -0,0 +1,55 @@
+import os
+
+from mypy_extensions import Arg
+from sgqlc.types import non_null, Variable
+
+from answer_rocket.auth import AuthHelper
+from answer_rocket.graphql.client import GraphQlClient
+from answer_rocket.graphql.schema import JSON, String, UUID
+from answer_rocket.output import ChatReportOutput
+from answer_rocket.graphql.schema import UUID as GQL_UUID
+
+
+class Skill:
+    """
+    Provides tools to interact with copilot skills directly.
+    """
+
+    def __init__(self, auth_helper: AuthHelper, gql_client: GraphQlClient):
+        self._auth_helper = auth_helper
+        self._gql_client = gql_client
+
+    def run(self, copilot_id: str, skill_name: str, parameters: dict | None = None) -> ChatReportOutput:
+        """
+        Runs a skill and returns its full output (does not stream intermediate skill output).
+
+        :param copilot_id: the id of the copilot to run the skill on
+        :param skill_name: the name of the skill to execute
+        :param parameters: a dict of parameters to pass to the skill where keys are the param keys and values are the values
+         to populate them with
+        :return the full output object of the skill
+        """
+
+        preview_query_args = {
+            "copilotId": UUID(copilot_id),
+            "skillName": skill_name,
+            'parameters': parameters or {},
+        }
+
+        preview_query_vars = {
+            'copilot_id': Arg(non_null(GQL_UUID)),
+            'skill_name': Arg(non_null(String)),
+            'parameters': Arg(JSON),
+        }
+
+        operation = self._gql_client.query(variables=preview_query_vars)
+
+        preview_query = operation.run_copilot_skill(
+            copilot_id=Variable('copilot_id'),
+            skill_name=Variable('skill_name'),
```

Comment:
do we technically enforce that these names are unique on the server side? I suspect we don't since each skill has a UUID we use internally, but it also wouldn't be the most friendly ux to ask them to go discover that UUID to use here. think it's fine if we use names here, but maybe we should consider making names unique within a copilot to prevent future headaches.

Reply:
Yep, that was a concern I had as well. We do appear to enforce name uniqueness already (at least from what I have seen on the frontend).